### PR TITLE
Allow mods to change class icons and class display names dynamically Issue #106

### PR DESCRIPTION
**Pull request by Musashi1584**
In order to allow mods to change class name, summary and icon dynamcly i propose to add two functions in `XComGameState_Unit`

```
GetSoldierClassIcon()
GetSoldierClassDisplayName()
GetSoldierClassSummary()
```

and change all old references to the the soldier class template to the new functions.

The new functions trigger events that allow to override the default.

### CODE REVIEW

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/UIChooseClass.uc [hunk 1]
```diff
@@ -89,8 +89,7 @@ simulated function array<Commodity> ConvertClassesToCommodities()
 	for (iClass = 0; iClass < m_arrClasses.Length; iClass++)
 	{
 		ClassTemplate = m_arrClasses[iClass];
-		
-		ClassComm.Title = ClassTemplate.DisplayName;
+		ClassComm.Title =  ClassTemplate.DisplayName;
```
Comment (robojumper) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/UIChooseClass.uc]: ?

Comment (Musashi1584) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/UIChooseClass.uc]: Will remove this class, i wanted to replace here but no unitstate available

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Unit.uc [hunk 2]
```diff
@@ -13640,6 +13640,56 @@ function bool UnitIsValidForPhotobooth()
...
+function String GetSoldierClassSummary()
+{
+	local XComLWTuple Tuple;
+
+	Tuple = new class'XComLWTuple';
+	Tuple.Id = 'SoldierClassSummary';
+	Tuple.Data.Add(1);
+
+	Tuple.Data[0].kind = XComLWTVString;
+	Tuple.Data[0].s = m_SoldierClassTemplate.ClassSummary;
```
Comment (robojumper) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Unit.uc]: Try not accessing m_SoldierClassTemplate directly in these three functions, it's a cached value that may be none for new or update Unit States where GetSoldierClassTemplate() hasn't been called yet.

Comment (Musashi1584) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Unit.uc]: done

**Comment by robojumper** [comment]
Looks good to me

**Pull request merged**